app/src/main/java/com/cupcake/chickenmasala/ui/fragment/home/adapter/HomeRecyclerAdapter.kt

```diff
@@ -0,0 +1,226 @@
+            is FilteredFoodViewHolder -> bindFilteredFood(holder, position)
+        }
+    }
+
+    private fun bindChipsFiltered(holder: ChipsViewHolder, position: Int) {
+        holder.binding.chipsFilter.setOnCheckedStateChangeListener { group, checkedIds ->
+            listener.onChipClicked(group, checkedIds[0])
+        }
+    }
+
+    private fun bindFilteredFood(holder: FilteredFoodViewHolder, position: Int) {
+        val recipe = items[position].item as Recipe
+        holder.binding.apply {
+            cuisineName.text = recipe.cuisine
+            recipeName.text = recipe.translatedRecipeName
+            recipeTime.text = recipe.totalTimeInMin.toString()
+            cardImage.setImage(recipe.imageUrl)
+            root.setOnClickListener {
+                listener.onCardClicked(recipe.id)
+            }
+        }
+    }
+
+    private fun bindRecentFood(holder: RecentFoodViewHolder, position: Int) {
+        val recentRecipes = items[position].item as List<Recipe>
+        val horizontalAdapter =
+            HorizontalRecipeRecyclerAdapter(listener)
+        horizontalAdapter.submitList(recentRecipes)
+        holder.binding.apply {
+            recyclerViewHorizontal.adapter = horizontalAdapter
+        }
+    }
+
+    private fun bindViewPager(holder: ViewPagerViewHolder, position: Int) {
+        val advices = items[position].item as List<HealthAdvice>
+        val viewPager = holder.binding.viewPager
+        val adapter = ViewPagerAdapter(viewPager, advices, listener)
+        holder.binding.apply {
+            viewPager.adapter = adapter
+            setupViewPager(viewPager, advices)
+            setUpTransformer(viewPager)
+            viewAll.setOnClickListener {
+                listener.onViewAllButtonClicked()
+            }
+
+        }
+    }
+
+    private fun ItemViewPagerBinding.setupViewPager(
+        viewPager: ViewPager2,
+        advices: List<HealthAdvice>
+    ) {
+        val runnable = Runnable {
+            viewPager.currentItem = viewPager.currentItem + 1
+        }
+
+        val handler = Handler(Looper.myLooper()!!)
+        viewPager.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+            override fun onPageSelected(position: Int) {
+                super.onPageSelected(position)
+                handler.removeCallbacks(runnable)
+                handler.postDelayed(runnable, 4000)
+            }
+        })
+        val viewPagerAdapter = ViewPagerAdapter(viewPager, advices, listener)
+        viewPagerAdapter.submitList(advices)
+
+        viewPager.adapter = viewPagerAdapter
+        viewPager.post(runnable)
+        viewPager.offscreenPageLimit = 3
+        viewPager.clipToPadding = false
+        viewPager.clipChildren = false
+        viewPager.getChildAt(0).overScrollMode = RecyclerView.OVER_SCROLL_NEVER
+
+    }
+
+    private fun ItemViewPagerBinding.setUpTransformer(viewPager: ViewPager2) {
```

Comment:
Defining extension functions inside a class in Kotlin is possible, but it may not always be the best approach. In general, it is recommended to define extension functions outside of the class they are extending.